src/Player.Vm.Api/Domain/Services/ViewService.cs

```diff
@@ -45,34 +48,88 @@ public ViewService(
 
         public async Task<Guid?> GetViewIdForTeam(Guid teamId, CancellationToken ct)
         {
-            Guid? viewId;
+            var teamInfo = await GetInfoForTeam(teamId, ct);
 
-            if (!_cache.TryGetValue(teamId, out viewId))
-            {
-                var team = await _playerApiClient.GetTeamAsync(teamId, ct);
-                viewId = team.ViewId;
+            return teamInfo.ViewId;
+        }
 
-                _cache.Set(teamId, viewId, new MemoryCacheEntryOptions().SetSlidingExpiration(TimeSpan.FromHours(12)));
+        public async Task<Guid[]> GetViewIdsForTeams(IEnumerable<Guid> teamIds, CancellationToken ct)
+        {
+            var teamInfoList = await GetInfoForTeams(teamIds, ct);
+            var viewIds = new List<Guid>();
+            foreach (var teamInfo in teamInfoList)
+            {
+                if (teamInfo.ViewId != null && !viewIds.Any(x => x == teamInfo.ViewId))
+                {
+                    viewIds.Add((Guid)teamInfo.ViewId);
+                }
             }
 
-            return viewId;
+            return viewIds.ToArray();
         }
 
-        public async Task<Guid[]> GetViewIdsForTeams(IEnumerable<Guid> teamIds, CancellationToken ct)
+        public async Task<TeamInfo[]> GetInfoForTeams(IEnumerable<Guid> teamIds, CancellationToken ct)
         {
-            var viewIds = new List<Guid>();
+            var teamInfoList = new List<TeamInfo>();
 
             foreach (var teamId in teamIds)
             {
-                var viewId = await this.GetViewIdForTeam(teamId, ct);
+                var teamInfo = await this.GetInfoForTeam(teamId, ct);
 
-                if (viewId.HasValue && !viewIds.Any(x => x == viewId.Value))
+                if (!teamInfoList.Any(x => x.ViewId == teamInfo.ViewId))
                 {
-                    viewIds.Add(viewId.Value);
+                    teamInfoList.Add(teamInfo);
                 }
             }
 
-            return viewIds.ToArray();
+            return teamInfoList.ToArray();
+        }
+
+        public async Task<List<Guid>> GetTeamsForView(Guid viewId, CancellationToken ct)
+        {
+            var teamIds = new List<Guid>();
+            if (!_cache.TryGetValue(viewId, out teamIds))
+            {
+                teamIds = (await _playerApiClient.GetViewTeamsAsync(viewId, ct)).Select(x => x.Id).ToList();
+                _cache.Set(viewId, teamIds, new MemoryCacheEntryOptions().SetSlidingExpiration(TimeSpan.FromHours(12)));
+            }
+
+            return teamIds;
         }
+
+        private async Task<TeamInfo> GetInfoForTeam(Guid teamId, CancellationToken ct)
+        {
+            TeamInfo teamInfo;
+
+            if (!_cache.TryGetValue(teamId, out teamInfo))
+            {
+                teamInfo = await GetTeamInfoFromPlayer(teamId, ct);
+                _cache.Set(teamId, teamInfo, new MemoryCacheEntryOptions().SetSlidingExpiration(TimeSpan.FromHours(12)));
```

Comment:
This should probably be a lot shorter now. Maybe 30 minutes or less? Previously, we were just saving the teamId to viewId mapping, which pretty much can't change. Now that you are saving the team and view names, you wouldn't want to use the old names for 12 hours after they were changed.

Ideally, we would update this automatically through webhooks in Player, but that's probably out of scope for this PR.

Reply:
I went to 15 minutes.  Still doesn't feel right.  When you make a change, you typically would like it to show up immediately, but I think 15 minutes is reasonable.